infra/Pos/Binary/Infra/Slotting.hs

```diff
@@ -4,24 +4,22 @@ module Pos.Binary.Infra.Slotting
        (
        ) where
 
+import           Universum
+
 import           Data.Time.Units    (Millisecond)
 
-import           Pos.Binary.Class   (Cons (..), Field (..), deriveSimpleBi)
+import           Pos.Binary.Class   (Cons (..), Field (..), Bi (..), Size( VarSize ), deriveSimpleBi, getSize)
 import           Pos.Binary.Core    ()
 import           Pos.Core.Timestamp (TimeDiff)
-import           Pos.Core.Types     (EpochIndex)
-import           Pos.Slotting.Types (EpochSlottingData (..), SlottingData (..))
+import           Pos.Slotting.Types (EpochSlottingData (..), SlottingData, getSlottingDataMap, createSlottingDataUnsafe)
 
 deriveSimpleBi ''EpochSlottingData [
     Cons 'EpochSlottingData [
         Field [| esdSlotDuration :: Millisecond |],
         Field [| esdStartDiff    :: TimeDiff    |]
     ]]
 
--- CSL-1122: add a test for serialization of 'SlottingData'
-deriveSimpleBi ''SlottingData [
-    Cons 'SlottingData [
-        Field [| sdPenult      :: EpochSlottingData |],
-        Field [| sdLast        :: EpochSlottingData |],
-        Field [| sdPenultEpoch :: EpochIndex        |]
-    ]]
+instance Bi SlottingData where
+    size = VarSize $ \slottingData -> getSize $ getSlottingDataMap slottingData
+    put slottingData = put $ getSlottingDataMap slottingData
+    get = createSlottingDataUnsafe <$> get
```

Comment:
I figured that since we can't create an instance outside `Pos.Slotting.Types` it wasn't possible to have it serialized (serialization presupposes creation). Even the `unsafe` constructor has some validation - look at `createSlottingDataUnsafe`.

Other than this and `createInitSlottingData` there isn't a way to construct it and I thought it was deserving a `error` (impossible state reconstructed).